solutions/USACO dec15-feast-gold.mdx

```diff
@@ -0,0 +1,50 @@
+---
+id: usaco-574
+source: USACO Gold 2015 December
+title: Fruit Feast
+author: Danh Ta Chi Thanh
+---
+
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_feast_gold_dec15.html)
+
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+bool dp[2][5000001];
+
+int main() {
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	freopen("feast.in","r",stdin);
+	freopen("feast.out","w",stdout);
+	int T,A,B;
+	cin >> T >> A >> B;
+	dp[0][0] = true;
+	dp[1][0] = true;
+	for (int i = 0; i < T + 1; ++i) {
+		if (i-A>=0) dp[0][i] |= dp[0][i-A];
+		if (i-B>=0) dp[0][i] |= dp[0][i-B];
+		dp[1][i/2] |= dp[0][i];
+	}
+	for (int i = 0; i < T + 1; ++i) {
+		if (i-A>=0) dp[1][i] |= dp[1][i-A];
+		if (i-B>=0) dp[1][i] |= dp[1][i-B];
+	}
+	for (int i = T; i >= 0; --i) if (dp[0][i] || dp[1][i]) {

```

Comment:
you don't need to check `dp[0][i]` here

Reply:
Oh, I get it, so in all cases where dp[0][i] is true, dp[1][i] is definitely true too?